### PR TITLE
Add Return the Page, Article, Snippet in ApplyWorkflow and CopyLocale Message Handlers

### PR DESCRIPTION
**Pull request by matthiasseghers**
# Handler Return Type Fix

| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes # 
| Related issues/PRs | # 
| License | MIT
| Documentation PR | sulu/sulu-docs# 

---

## What's in this PR?

This PR fixes an inconsistency in the return types of content MessageHandlers where some handlers (`ApplyWorkflowTransitionPageMessageHandler`, `CopyLocalePageMessageHandler`, `MovePageMessageHandler`, `OrderPageMessageHandler`) returned `void` while other content modification handlers returned their respective interface (`PageInterface`, `ArticleInterface`, `SnippetInterface`).

### Problem

When working with Sulu 3.0's message-based architecture, there was an inconsistency in handler return types across content types:

**Page Handlers that RETURNED PageInterface:**
- `CreatePageMessageHandler` - Returns the created page
- `ModifyPageMessageHandler` - Returns the modified page
- `CopyPageMessageHandler` - Returns the copied page
- `RestorePageVersionMessageHandler` - Returns the restored page

**Page Handlers that RETURNED void:**
- `ApplyWorkflowTransitionPageMessageHandler` - Should return the page with updated workflow state
- `CopyLocalePageMessageHandler` - Should return the page with new locale content
- `MovePageMessageHandler` - Should return the page after moving
- `OrderPageMessageHandler` - Should return the page after reordering

**Similar inconsistencies existed in Article and Snippet packages.**

This inconsistency created several issues:

1. **Cannot verify operations succeeded** - After publishing/unpublishing/moving/ordering, there's no way to get the updated state without refetching
2. **Cannot get updated entity state** - Must use workarounds like `EntityManager::refresh()` or refetching
3. **API inconsistency** - Similar operations (modifying content state) have different return types across packages
4. **Difficult to chain operations** - Cannot build fluent APIs when some handlers return void

### Real-World Impact

```php
// Current problem: Can't verify publish succeeded
$this->handle(new ApplyWorkflowTransitionPageMessage(['uuid' => $uuid], 'en', 'publish'));
// Did it publish? Unknown! Must refetch:
$page = $this->pageRepository->findPage($uuid, 'en');
$content = $this->resolvePage($page, 'en');
if ($content->getWorkflowPlace() === 'published') {
    // Now we know it worked
}
```

### Solution

By returning the respective interface from these handlers, callers can immediately access the updated entity state without additional database queries, making the API consistent across all content types (Page, Article, Snippet).

### Page - Apply Workflow Transition

**Before (Current - Must Refetch)**
```php
use Sulu\Page\Application\Message\ApplyWorkflowTransitionPageMessage;
use Sulu\Content\Domain\Model\WorkflowInterface;

$message = new ApplyWorkflowTransitionPageMessage(
    ['uuid' => $pageUuid],
    'en',
    WorkflowInterface::WORKFLOW_TRANSITION_PUBLISH
);

$this->handle($message);  // Returns void

// Can't verify directly - must refetch
$page = $this->pageRepository->findPage($pageUuid, 'en');
$content = $this->resolvePage($page, 'en');
```

**After (With This PR - Direct Return)**
```php
use Sulu\Page\Application\Message\ApplyWorkflowTransitionPageMessage;
use Sulu\Content\Domain\Model\WorkflowInterface;

$message = new ApplyWorkflowTransitionPageMessage(
    ['uuid' => $pageUuid],
    'en',
    WorkflowInterface::WORKFLOW_TRANSITION_PUBLISH
);

$page = $this->handle($message);  // Returns PageInterface

// Can verify immediately - no refetch needed
$content = $this->resolvePage($page, 'en');
```

### Page - Move Page

**Before**
```php
$message = new MovePageMessage(
    ['uuid' => $pageUuid],
    ['uuid' => $destinationParentUuid],
    'en'
);
$this->handle($message);  // Returns void
// Must refetch to verify move succeeded
```

**After**
```php
$message = new MovePageMessage(
    ['uuid' => $pageUuid],
    ['uuid' => $destinationParentUuid],
    'en'
);
$page = $this->handle($message);  // Returns PageInterface
// Can verify immediately
```

### Article - Copy Locale

**Before**
```php
$message = new CopyLocaleArticleMessage(['uuid' => $uuid], 'en', 'de');
$this->handle($message);  // Returns void
// Must refetch to access DE content
$article = $this->articleRepository->findArticle($uuid, 'de');
```

**After**
```php
$message = new CopyLocaleArticleMessage(['uuid' => $uuid], 'en', 'de');
$article = $this->handle($message);  // Returns ArticleInterface
// DE content immediately available
$deContent = $this->resolveArticle($article, 'de');
```

### Chaining Operations

```php
// Create and immediately publish in one flow
$page = $this->handle(new CreatePageMessage($webspace, $parent, [
    'locale' => 'en',
    'title' => 'New Page',
    'template' => 'default',
]));

// Chain: Now publish it
$publishedPage = $this->handle(new ApplyWorkflowTransitionPageMessage(
    ['uuid' => $page->getUuid()],
    'en',
    WorkflowInterface::WORKFLOW_TRANSITION_PUBLISH
));

// Both operations return PageInterface - clean, consistent API
```

## Handler Coverage

All content modification handlers now consistently return their respective interface:

### Page Handlers ✅
- `CreatePageMessageHandler` → `PageInterface`
- `ModifyPageMessageHandler` → `PageInterface`
- `CopyPageMessageHandler` → `PageInterface`
- `RestorePageVersionMessageHandler` → `PageInterface`
- `ApplyWorkflowTransitionPageMessageHandler` → `PageInterface` ✨ **UPDATED**
- `CopyLocalePageMessageHandler` → `PageInterface` ✨ **UPDATED**
- `MovePageMessageHandler` → `PageInterface` ✨ **UPDATED**
- `OrderPageMessageHandler` → `PageInterface` ✨ **UPDATED**

### Article Handlers ✅
- `CreateArticleMessageHandler` → `ArticleInterface`
- `ModifyArticleMessageHandler` → `ArticleInterface`
- `RestoreArticleVersionMessageHandler` → `ArticleInterface`
- `ApplyWorkflowTransitionArticleMessageHandler` → `ArticleInterface` ✨ **UPDATED**
- `CopyLocaleArticleMessageHandler` → `ArticleInterface` ✨ **UPDATED**

### Snippet Handlers ✅
- `CreateSnippetMessageHandler` → `SnippetInterface`
- `ModifySnippetMessageHandler` → `SnippetInterface`
- `RestoreSnippetVersionMessageHandler` → `SnippetInterface`
- `ApplyWorkflowTransitionSnippetMessageHandler` → `SnippetInterface` ✨ **UPDATED**
- `CopyLocaleSnippetMessageHandler` → `SnippetInterface` ✨ **UPDATED**

*Note: Deletion handlers (`RemovePageMessageHandler`, `RemoveArticleMessageHandler`, etc.) correctly remain `void` as they don't return entities.*

## Breaking Changes

**No breaking changes by this change.**

1. **Existing code continues to work** - Code that doesn't capture the return value still functions
2. **No method signature changes** - Only the return type is added (void → Interface)
3. **Same exceptions thrown** - Error handling behavior unchanged
4. **No changes to message structure** - Message classes remain the same

## To Do

- [ ] Create a documentation PR
- [ ] ~~Add breaking changes to UPGRADE.md~~ (Not needed - backward compatible)

### CODE REVIEW

**Comment by alexander-schranz** [comment]
Thx for the pull request, this should be consistent so we should adopt the same for Article and Snippets also.

File: packages/page/src/Application/MessageHandler/CopyLocalePageMessageHandler.php [hunk 1]
```diff
@@ -59,11 +60,14 @@ public function __invoke(CopyLocalePageMessage $message): void
...
-            [],
-        ));
+        $this->domainEventCollector->collect(
+            new PageTranslationCopiedEvent(
+                $page,
+                $message->getSourceLocale(),
+                $message->getTargetLocale()
+            )
+        );
+
```
Comment (alexander-schranz) [inline packages/page/src/Application/MessageHandler/CopyLocalePageMessageHandler.php]: unexpected change which make the CI fail:

```suggestion
        $this->domainEventCollector->collect(new PageTranslationCopiedEvent(
            $page,
            $message->getTargetLocale(),
            $message->getSourceLocale(),
            [],
        ));
        
```

**Comment by Prokyonn** [comment]
@matthiasseghers to have this consistent, I think we should also add the return type to the `MovePageMessageHandler`, `OrderPageMessageHandler` and the `RemovePageTranslationMessageHandler`

I think it makes sense for all MessageHandlers excluding the `RemovePageMessageHandler`. 

Same applies to the Snippet and Article messagehandlers

**Comment by matthiasseghers** [comment]
@Prokyonn  I think that `RemovePageTranslationMessageHandler` is still fine to have void, as both `RemovePageTranslationMessageHandler` and `RemovePageMessageHandler` handle a delete action - which is fine to have a void being returned. The others are creation and or update so I'll make the changes to the OrderPage and MovePage as well for consistency.

**Comment by Prokyonn** [comment]
> @Prokyonn I think that `RemovePageTranslationMessageHandler` is still fine to have void, as both `RemovePageTranslationMessageHandler` and `RemovePageMessageHandler` handle a delete action - which is fine to have a void being returned. The others are creation and or update so I'll make the changes to the OrderPage and MovePage as well for consistency.

`RemovePageTranslationMessageHandler` does only remove one locale from the page, therefore a PageDimensionContent entity is removed, but the `Page` entity itself still exists, so I wouldn't mind returning it also

**Comment by matthiasseghers** [comment]
> > @Prokyonn I think that `RemovePageTranslationMessageHandler` is still fine to have void, as both `RemovePageTranslationMessageHandler` and `RemovePageMessageHandler` handle a delete action - which is fine to have a void being returned. The others are creation and or update so I'll make the changes to the OrderPage and MovePage as well for consistency.
> 
> `RemovePageTranslationMessageHandler` does only remove one locale from the page, therefore a PageDimensionContent entity is removed, but the `Page` entity itself still exists, so I wouldn't mind returning it also

What is the behaviour in the case of using RemovePageTranslationMessageHandler on a page with only one translation available? Will it fail? Not do anything? Or delete the page?

**Review by alexander-schranz — APPROVED** [review]

**Comment by Prokyonn** [comment]
> > > @Prokyonn I think that `RemovePageTranslationMessageHandler` is still fine to have void, as both `RemovePageTranslationMessageHandler` and `RemovePageMessageHandler` handle a delete action - which is fine to have a void being returned. The others are creation and or update so I'll make the changes to the OrderPage and MovePage as well for consistency.
> > 
> > 
> > `RemovePageTranslationMessageHandler` does only remove one locale from the page, therefore a PageDimensionContent entity is removed, but the `Page` entity itself still exists, so I wouldn't mind returning it also
> 
> What is the behaviour in the case of using RemovePageTranslationMessageHandler on a page with only one translation available? Will it fail? Not do anything? Or delete the page?

I’m not sure what will happen in this edge case, and I doubt our tests currently cover it 😅. What I do know is that the admin UI prevents removing a locale when only one exists.
However, if you dispatch the message anyway, the last DimensionContent will be removed, leaving the  Page  without any DimensionContents . I’m not certain whether this state would cause issues in the admin UI.

**Review by Prokyonn — CHANGES_REQUESTED** [review]
RemoveSnippetTranslationMessageHandler and RemoveArticleTranslationMessageHandler is still missing the return type. Rest looks good 👍

**Comment by alexander-schranz** [comment]
@Prokyonn tackling this in upcoming pull request. Not able to push into this branch as it a default branch on @matthiasseghers fork which don't allow maintainers to push.
